test/mocks/overrides/override_firebase_transaction.dart

```diff
@@ -0,0 +1,38 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+
+/// A mock class of a cloud firestore [Transaction].
+/// This mock will always throw the [FirebaseException] error
+/// when any of its method is called.
+/// It is used to test the offline error handling of our services.
+class MockErrorFirebaseTransaction implements Transaction {
+  static const _plugin = "mock_error_firebase_transaction";
```

Comment:
```suggestion
  static const _plugin = "mockErrorFirebaseTransaction";
```
We generally use camel case in this project. Should we do the same here?

Reply:
I just followed Firestore FlutterFire's own coding style conventions for naming plugins. You can see for yourself in the original error I pasted above, it comes from the plugin `cloud_firestore/unavailable`.

I think we should leave it like that.

Reply:
Yes no problem it is completely justified 👍🏼